src/binary.ts

```diff
@@ -292,3 +293,204 @@ export class Binary {
 }
 
 Object.defineProperty(Binary.prototype, '_bsontype', { value: 'Binary' });
+
+/** @public */
+export type UUIDExtended = {
+  $uuid: string;
+};
+const BYTE_LENGTH = 16;
+
+const kId = Symbol('id');
+
+/**
+ * A class representation of the BSON UUID type.
+ * @public
+ */
+export class UUID extends Binary {
+  static cacheHexString: boolean;
+
+  /** UUID Bytes @internal */
+  private [kId]!: Buffer;
```

Comment:
Is this needed, given that the parent class has a `buffer` property that provides the same information?